pkg/pillar/vendor/github.com/digitalocean/go-qemu/qmp/socket.go

```diff
@@ -135,13 +135,29 @@ func (mon *SocketMonitor) Connect() error {
 		return err
 	}
 
-	// Check for no error on return
-	var r response
-	if err := dec.Decode(&r); err != nil {
-		return err
-	}
-	if err := r.Err(); err != nil {
-		return err
+	// Wait for response
+	scanner := bufio.NewScanner(mon.c)
+	for scanner.Scan() {
+		var e Event
+
+		b := scanner.Bytes()
+		if err := json.Unmarshal(b, &e); err != nil {
+			return err
+		}
+
+		// If not an event then our qmp capabilities response
+		if e.Event == "" {
+			var r response
+			if err := json.Unmarshal(b, &r); err != nil {
+				return err
+			}
+			// Check response on errors
+			if err := r.Err(); err != nil {
+				return err
+			}
+			break
+		}
+		// Drop possible event, continue reading
```

Comment:
Are we potentially overlooking important information by discarding events before the capability response?

Reply:
what important information?

Reply:
Doesn't make sense to at least log discarded events?

Reply:
I don't know... Is it possible that another entity is waiting for the events that we are dropping? I guess the answer is no, but it's better to ask :)

Reply:
All events will be received on the listener QMP socket we have (see the `qmpEventHandler()`),  Here in the `Connect()`  we are not interested in events until a valid response is received.